usr/share/rear/lib/layout-functions.sh

```diff
@@ -611,3 +611,207 @@ retry_command ()
+                ;;
+            *mapper[/!]*)
+                case $OS_VENDOR in
+                    SUSE_LINUX)
+                    # SUSE Linux SLE12 put a "-part" between [mpath device name] and [part number].
+                    # For example /dev/mapper/3600507680c82004cf8000000000000d8-part1.
+                    # But SLES11 uses a "_part" instead. (Let's assume it is the same for SLES10 )
+                    if (( $OS_VERSION < 12 )) ; then
+                        # For SUSE before version 12
+                        part_base="${original}_part" # append _part between main device and partitions
+                    else
+                        # For SUSE 12 or above
+                        part_base="${original}-part" # append -part between main device and partitions
+                    fi
+                    ;;
+                    RedHatEnterpriseServer)
+                        # RHEL 7 and above seems to named partitions on multipathed devices with
+                        # [mpath device name] + [part number] like standard disk.
+                        # For example: /dev/mapper/mpatha1
+
+                        # But the scheme in RHEL 6 need a "p" between [mpath device name] and [part number].
+                        # For exemple: /dev/mapper/mpathap1
+                        if (( $OS_VERSION < 7 )) ; then
+                            part_base="${original}p" # append p between main device and partitions
+                        fi
+                    ;;
+                    Ubuntu)
+                        # Ubuntu 16.04 (need to check for other version) named muiltipathed partitions with
+                        # [mapth device name] + "-part" + [part number]
+                        # for example : /dev/mapper/mpatha-part1
+                        part_base="${original}-part" # append -part between main device and partitions
+                    ;;
+
+                esac
+            ;;
+        esac
+        sed -i -r "\|$original|s|${part_base}([0-9]+)|$replacement\1|g" "$file_to_migrate"
+        # Replace whole devices
+        ### note that / is a word boundary, so is matched by \<, hence the extra /
+        sed -i -r "\|$original|s|/\<${original#/}\>|${replacement}|g" "$file_to_migrate"
+    done < "$replacement_file"
+
+    # Replace all replacements with their target.
+    while read source target junk ; do
+        replacement=$(get_replacement "$source")
+        # Replace whole device
+        sed -i -r "\|$replacement|s|$replacement\>|$target|g" "$file_to_migrate"
+        # Replace partitions
+        case "$target" in
+            *mmcblk[0-9]|*nvme[0-9]*n[1-9]|*rd[/!]c[0-9]d[0-9]|*cciss[/!]c[0-9]d[0-9]|*ida[/!]c[0-9]d[0-9]|*amiraid[/!]ar[0-9]|*emd[/!][0-9]|*ataraid[/!]d[0-9]|*carmel[/!][0-9])
+                target="${target}p" # append p between main device and partitions
+                ;;
+            *mapper[/!]*)
+                case $OS_VENDOR in
```

Comment:
Same as above. Since this is the same logic, could you extract that into a function like `format_device_name` or something like this? Makes it also easier to keep it in sync if something changes.

Reply:
done. I've named it `get_part_device_name_format()`... it is a bit long, but `format_device_name` sound a bit "dangerous" and make me nervous :) (=> formatting a device ????)

Reply:
@schabrolles 
don't worry when meaningful names become longer.
But worry about meaningful and unambiguous names
(unambiguous also means names that are not misleading)
cf. https://github.com/rear/rear/wiki/Coding-Style